aptos-move/framework/aptos-framework/sources/configs/Stake.move

```diff
@@ -278,25 +452,24 @@ module AptosFramework::Stake {
     /// 1. distribute rewards to active/pending_inactive delegations
     /// 2. process pending_active, pending_inactive correspondingly
     /// This function shouldn't abort.
-    fun update_stake_pool(addr: address) acquires StakePool {
+    fun update_stake_pool(addr: address) acquires StakePool, TestCoinCapabilities {
         let stake_pool = borrow_global_mut<StakePool>(addr);
         distribute_reward( &mut stake_pool.active);
         distribute_reward( &mut stake_pool.pending_inactive);
+
         // move pending_active to active
-        IterableTable::append(&mut stake_pool.active, &mut stake_pool.pending_active);
-        // move pending_inactive to inactive
-        IterableTable::append(&mut stake_pool.inactive, &mut stake_pool.pending_inactive);
-    }
-
-    /// Mint the reward and add to the delegation based on some formula
-    fun distribute_reward(v: &mut IterableTable<address, Delegation>) {
-        let key = IterableTable::head_key(v);
-        while (Option::is_some(&key)) {
-            let (delegation, _, next) = IterableTable::borrow_iter_mut(v, *Option::borrow(&key));
-            let reward = Coin::zero<TestCoin>(); // mint some coins based on delegation, timestamp, maybe also total stakes
-            Coin::merge<TestCoin>(&mut delegation.rewards, reward);
-            key = next;
-        };
+        Coin::merge<TestCoin>(&mut stake_pool.active, Coin::extract_all<TestCoin>(&mut stake_pool.pending_active));
+        Coin::merge<TestCoin>(&mut stake_pool.inactive, Coin::extract_all<TestCoin>(&mut stake_pool.pending_inactive));
+    }
+
+    /// Distribute reward corresponding to `stake`. Rewards is currently simple and just a fixed % of staked coins.
+    fun distribute_reward(stake: &mut Coin<TestCoin>) acquires TestCoinCapabilities {
```

Comment:
It might make sense to make this a native function that can look at the history of all voting rounds within the epoch. We'll likely need to make some form of indexed data to support this :/

Reply:
This is something I'll think through a bit more.

Reply:
Updated now to support performance-based rewards.

Reply:
I reverted the complex reward changes for now. Will do that in a separate followup PR.